mongo/src/main/java/com/redhat/lightblue/mongo/crud/IterateAndUpdate.java

```diff
@@ -186,6 +200,16 @@ public void update(CRUDOperationContext ctx,
                 measure.begin("ctx.addDocument");
                 DocTranslator.TranslatedDoc translatedDoc=translator.toJson(document);
                 DocCtx doc=new DocCtx(translatedDoc.doc,translatedDoc.rmd);
+                if (memoryMonitor != null) {
+                    // if memory threshold is exceeded, this will throw an Error
+                    memoryMonitor.apply(doc);
```

Comment:
Write and read limits are set separately. We can enforce a lower limit on finds than on updates, we can even disable the limit for updates completely.

Inconsistent state as a result of a failed update of a range of documents is not a new problem, it's just that this logic can make it happen more often.